### DISCUSSION TITLE
Converting prodigy jsonl to spacy: am I missing something?

### DISCUSSION THREAD

**Discussion by Zatteliet** [explosion/spaCy#10202] [Help: Coding & Implementations] [answered]
Supposing I have a set of documents annotated with Prodigy, so they are in the output format as in [examples in the docs](https://prodi.gy/docs/api-interfaces). I want to convert these to Spacy `Doc`s.

So far I've been doing this manually, with e.g.

```
def jsonl2doc(jsonl_dict: dict, nlp: Language) -> Doc:
    doc = nlp(jsonl_dict["text"])
    ents = []
    for ann in jsonl_dict.get("spans", []):
        span = Span(
            doc,
            ann["token_start"],
            ann["token_end"] + 1,
            ann["label"],
        )
        ents.append(span)
    doc.set_ents(ents)
    return doc
```

It works, but it feels like either prodigy or spacy is missing some built-in functionality around this, given how tightly the libraries are couples otherwise. Is there for obvious way of solving this task that I'm missing?

**Comment by thomashacker** [comment, marked as answer]
Hey,
If you annotated the data with prodigy, you can use the `data-to-spacy` command (https://prodi.gy/docs/recipes#data-to-spacy) to get the `.spacy` files for training.
